icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -1882,7 +1882,7 @@ void RBBIMonkeyKind::setText(const UnicodeString &s) {
             printf("Failed to resolve at %zu between U+%04X and U+%04X ", i, s.char32At(i - 1),
                    s.char32At(i));
             if (resolved[i].indexInRemapped.has_value()) {
-                printf("which is remapped %zu between U+%04X and U+%04X", *resolved[i].indexInRemapped,
+                printf("which is remapped %d between U+%04X and U+%04X", *resolved[i].indexInRemapped,
```

Comment:
This probably works, but if it still causes trouble on some platform, then you could cast the index to `int` so that it's a guaranteed match with `%d`.